weather-app/src/utils/index.js

```diff
@@ -0,0 +1,44 @@
+export const initAutocomplete = (input, callback) => {
+  const autocomplete = new google.maps.places.Autocomplete(input, { types: ['(cities)'] });
+  autocomplete.addListener('place_changed', () => callback(autocomplete.getPlace()));
+};
+
+export const bindAll = (context, ...names) => {
+  names.forEach(name => {
+    if (typeof context[name] === 'function') {
+      context[name] = context[name].bind(context);
+    } else {
+      throw Error(
+        `Expected function ${name}. Instead received: ${typeof context[name]}`
+      );
+    }
+  });
+};
+
+export const insert = (host, children) => {
+  const node = host;
+  if (typeof children === 'string') {
+    node.insertAdjacentHTML('beforeend', children)
+  } else if (Array.isArray(children)) {
+      children.forEach(elem => {
+        (typeof elem === 'string') ? node.insertAdjacentHTML('beforeend', elem) : node.append(elem);
+      });
+  } else {
+    node.append(elem);
+  };
+  return node;
+}
+
+const removeDuplicates = (myArr, prop) => {
+  return myArr.filter((obj, pos, arr) => {
+      return arr.map(mapObj => mapObj[prop]).indexOf(obj[prop]) === pos;
+  });
+};
+
+export const addToArray = (list, location) => {
```

Comment:
If you need only uniq elements in array, probably, you need to use `Set`? 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set